index.js

```diff
@@ -4,23 +4,41 @@ var path = require('path')
 var PdfPrinter = require('pdfmake')
 var streamToArray = require('stream-to-array')
 
-var defaultFonts = {
-  Roboto: {
-    normal: path.join(__dirname, '/fonts/Roboto-Regular.ttf'),
-    bold: path.join(__dirname, '/fonts/Roboto-Medium.ttf'),
-    italics: path.join(__dirname, '/fonts/Roboto-Italic.ttf'),
-    bolditalics: path.join(__dirname, '/fonts/Roboto-MediumItalic.ttf')
-  }
-}
+var defaultFonts = {}
 
-function Document (docDefinition) {
 function Document (docDefinition, renderOptions) {
   this.docDefinition = docDefinition
   this.renderOptions = renderOptions
 }
 
 Document.prototype._createDoc = function (options) {
   options = options || {}
+
+  var userLanguage = this.renderOptions.userLanguage
+
+// pdfMake requires the objects defaultFonts and Roboto to be passed to it regardless of what actual fonts are used. This way we don't have to add any custom fonts to the pdfmake dependency directly. 
+  if (userLanguage === 'zh' || userLanguage === 'ko') {
+    defaultFonts = {
+      Roboto: {
+        //Source Han Sans supports Latin orthography + CJK but not Cyrillic or Greek 
+        normal: path.join(__dirname, '/fonts/SourceHanSans-Normal.ttf'),
+        bold: path.join(__dirname, '/fonts/SourceHanSans-Medium.ttf'),
+        italics: path.join(__dirname, '/fonts/Roboto-Italic.ttf'),
+        bolditalics: path.join(__dirname, '/fonts/Roboto-MediumItalic.ttf')
+      }
+    }
+  } else {
+    defaultFonts = {
+        Roboto: {
+          //Roboto supports all major European orthographies but not CJK
+          normal: path.join(__dirname, '/fonts/Roboto-Regular.ttf'),
+          bold: path.join(__dirname, '/fonts/Roboto-Medium.ttf'),
+          italics: path.join(__dirname, '/fonts/Roboto-Italic.ttf'),
+          bolditalics: path.join(__dirname, '/fonts/Roboto-MediumItalic.ttf')
+        }
+      }
+    }
```

Comment:
You are setting the global `defaultFonts` variable in `_createDoc`. So the value depends on the parameters of the last `_createDoc` call, which is a weird side effect. 
I think you should either only declare the variable inside the `_createDoc` function or just use multiple variables:

```js
var sourceHans = { defaultFonts: {...} }
var robotto = { defaultFonts: {...} }

```